cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -423,7 +440,51 @@ func (machinedeployment *MachineDeployment) Nodes() ([]cloudprovider.Instance, e
 // Implementation optional.
 // TODO: add proper implementation
 func (machinedeployment *MachineDeployment) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	mcdSpec, err := machinedeployment.mcmManager.GetMachineDeploymentSpec(machinedeployment.Name)
+	if err != nil {
+		return nil, err
```

Comment:
Returning an error means that each option will be assigned a zero value (See the caller of this method to confirm). I don't think we want that. We can add a log saying that we failed to get the machine deployment and hence will use the defaults for this node group. What do you think @himanshu-kun?

Reply:
Just want to confirm if I understood this right. If there is an error, is the suggestion to return `(default_values, nil)` instead of `(nil, err)`?

Reply:
Ok. I rechecked the code. If an error is returned from `GetOptions`, its caller in `node_group_config_processor.go` also returns the error and then it gets logged as well. I think this is fine. Sorry for the confusion. No change is needed.